src/NuGet.Server.Core/Infrastructure/ServerPackage.cs

```diff
@@ -249,5 +253,41 @@ private static List<PackageDependencySet> ParseDependencySet(string value)
 
             return Tuple.Create(id, versionSpec, targetFramework);
         }
+
+        private static bool IsPackageSemVer2(IPackage package)
```

Comment:
This is exactly the method that we may need to re-use in gallery and search service/jobs. Is the intention for us to all consume NuGet.Server.Core (as a NuGet package)?